llvm/lib/CodeGen/AsmPrinter/AsmPrinter.cpp

```diff
@@ -3632,7 +3663,8 @@ static void emitGlobalConstantCHERICap(const DataLayout &DL, const Constant *CV,
   } else if (const MCSymbolRefExpr *SRE = dyn_cast<MCSymbolRefExpr>(Expr)) {
     if (auto BA = dyn_cast<BlockAddress>(CV)) {
       // For block addresses we emit `.chericap FN+(.LtmpN - FN)`
-      auto FnStart = AP.getSymbol(BA->getFunction());
+      // NB: Must use a non-preemptible symbol
+      auto FnStart = AP.getSymbolPreferLocal(*BA->getFunction(), true);
```

Comment:
Nice, I was going to make that change but didn't get around to the code to determine whether to emit the label.